### PR TITLE
Replacing strftime with std::time_put

### PR DESCRIPTION
**Pull request by phprus**
Continuation of PR #2544.

~~Draft, because I haven't yet tested the PR in all my environments.~~

@vitaut, what do you think about using the built-in month and weekday names in the case of the ``std::locale::classic()`` locale (without ``std::time_put`` calls)?

### CODE REVIEW

**Comment by vitaut** [comment]
> what do you think about using the built-in month and weekday names in the case of the std::locale::classic() locale (without std::time_put calls)?

We should definitely use built-in names in this case.

File: include/fmt/chrono.h [hunk 1]
```diff
@@ -1561,17 +1575,51 @@ template <typename OutputIt, typename Char> class tm_writer {
...
+  }
 
   void on_datetime(numeric_system ns) {
-    format_localized('c', ns == numeric_system::standard ? '\0' : 'E');
+    if (is_classic_) {
+#ifdef _WIN32
+      on_us_date();
+      *out_++ = ' ';
+      on_iso_time();
+#else
```
Comment (vitaut) [inline include/fmt/chrono.h:1597]: The locale-independent and C locale format should be consistent among platforms so let's remove this.

File: include/fmt/chrono.h [hunk 2]
```diff
@@ -1685,7 +1733,22 @@ template <typename OutputIt, typename Char> class tm_writer {
     write2(tm_sec());
   }
 
-  void on_12_hour_time() { format_localized('r'); }
+  void on_12_hour_time() {
+    if (is_classic_) {
+#ifdef _WIN32
+      on_iso_time();
+#else
```
Comment (vitaut) [inline include/fmt/chrono.h:1740]: Let's remove this as well.

**Review by vitaut — CHANGES_REQUESTED** [review]
Looks great, just minor comments left.

File: include/fmt/chrono.h (hunk 1 above)
Comment (phprus) [reply include/fmt/chrono.h]: Done. Corresponding tests for Windows are disabled (https://github.com/fmtlib/fmt/runs/4055476293?check_suite_focus=true)

File: test/chrono-test.cc [hunk 3]
```diff
@@ -245,13 +248,18 @@ TEST(chrono_test, time_point) {
...
-      "%r",  "%R",  "%T",  "%p",  "%z",  "%Z"};
+      "%%",  "%n",  "%t",  "%Y",  "%EY", "%y",  "%Oy", "%Ey", "%C",
+      "%EC", "%G",  "%g",  "%b",  "%h",  "%B",  "%m",  "%Om", "%U",
+      "%OU", "%W",  "%OW", "%V",  "%OV", "%j",  "%d",  "%Od", "%e",
+      "%Oe", "%a",  "%A",  "%w",  "%Ow", "%u",  "%Ou", "%H",  "%OH",
+      "%I",  "%OI", "%M",  "%OM", "%S",  "%OS", "%x",  "%Ex", "%X",
+      "%EX", "%D",  "%F",  "%R",  "%T",  "%p",  "%z",  "%Z"};
   spec_list.push_back("%Y-%m-%d %H:%M:%S");
+#ifndef _WIN32
+  // Disabled on Windows, because this formats is not consistent among platforms
```
Comment (vitaut) [inline test/chrono-test.cc:259]: this -> these and please add punctuation (.) at the end of the sentence.

**Comment by vitaut** [comment]
Thank you!

**Pull request merged**